api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -154,8 +152,16 @@ private DbUser initializeUserIfNeeded() {
       // If the user is already registered, their profile will get updated.
       fireCloudService.registerUser(dbUser.getGivenName(), dbUser.getFamilyName());
 
+      // by approving the latest AOU Terms of Service, the user has also approved the Terra TOS
+      try {
+        profileService.validateTermsOfService(dbUser);
+        fireCloudService.acceptTermsOfService();
+      } catch (BadRequestException e) {
+        // TODO 7834
+      }
+
       dbUser.setFirstSignInTime(new Timestamp(clock.instant().toEpochMilli()));
-      return saveUserWithConflictHandling(dbUser);
+      dbUser = saveUserWithConflictHandling(dbUser);
```

Comment:
This doesn't appear to change the behavior of the function. Am I wrong?

Reply:
you are right.  reverted.